.github/actions/verify-badge/action.yaml

```diff
@@ -0,0 +1,257 @@
+name: Verify-Badge
+description: Perform verification to assist with UDS badge certification
+
+runs:
+  using: composite
+  steps:
+    - name: Test installing the package
+      shell: bash
+      run: |
+        set -uo pipefail
+
+        # Validate UDS Package/Config chart is valid
+
+        echo "ℹ️ Installing kubectl-validate..."
+        go install sigs.k8s.io/kubectl-validate@latest
+
+        echo "ℹ️ Validating uds-config chart..."
+        kubectl validate <(helm template chart chart/)
+        echo
+
+        # Setup
+
+        NAMESPACE=$(yq '.components[].charts[].namespace' common/zarf.yaml | uniq)
+        echo "ℹ️ Namespace: $NAMESPACE"
+
+        UDS_PACKAGE_JSON=$(kubectl get Packages -n "${NAMESPACE}" -o jsonpath='{.items[]}')
+        echo "ℹ️ Retrieved UDS Package JSON"
+
+        UDS_PACKAGE_NAME=$(echo "$UDS_PACKAGE_JSON" | jq -r '.metadata.name')
+        echo "ℹ️ Package Name: $UDS_PACKAGE_NAME"
+        echo
+
+        # Istio
+
+        ## Must define any external interfaces under the expose key.
+
+        ENDPOINTS="$(echo "$UDS_PACKAGE_JSON" | jq -r '.status.endpoints[]')"
+        echo "ℹ️ Endpoints: $ENDPOINTS"
+
+        for ENDPOINT in $ENDPOINTS; do
+          # Curl endpoint and check that status isn't 404 or 5XX
+          STATUS=$(curl -s -o /dev/null -w "%{http_code}" "https://$ENDPOINT")
+          if [ "$STATUS" -eq 404 ] || [[ $STATUS == 5* ]]; then
+            echo "❌ Endpoint $ENDPOINT is returning $STATUS"
+            exit 1
+          fi
+          echo "✅ Endpoint $ENDPOINT was successfully curl'd"
+        done
+
+        echo
+
+        ## Must deploy and operate successfully with Istio injection enabled in the namespace.
+
+        POD_COUNT=$(kubectl get pods -n "${NAMESPACE}" --no-headers | wc -l)
+
+        POD_SIDECAR_COUNT=$(kubectl get pods -n "${NAMESPACE}" -o json | jq '.items[].spec.containers[] | select(.name=="istio-proxy") | length' | wc -l)
+
+        if [ "$POD_COUNT" -ne "$POD_SIDECAR_COUNT" ]; then
+          echo "❌ Not all pods have the istio sidecar"
+          exit 1
+        fi
+        echo "✅ All pods have the istio sidecar"
+
+        echo
+
+        ## Should avoid workarounds such as disabling strict mTLS peer authentication.
+
+        PEERAUTH=$(kubectl get Peerauthentication -n "${NAMESPACE}" -o=json | jq -r '.items[].spec.mtls.mode')
+
+        if [ "$PEERAUTH" != "STRICT" ] && [ "$PEERAUTH" != "" ]; then
+          echo "❌ Peerauth is not strict"
+          exit 1
```

Comment:
we may not want to hard-exit on Shoulds - also this is tricky since some packages (like GitLab) place the PeerAuthentication exceptions in _other_ namespaces. https://github.com/defenseunicorns/uds-package-gitlab/blob/main/chart/templates/postgres-peerauthentication.yaml#L6

Reply:
There's a few other shoulds in the SSO section - agreed that they should not fail.

Reply:
fixed the shoulds to be warnings